burn-autodiff/src/tests/mask.rs

```diff
@@ -23,4 +23,32 @@ mod tests {
         assert_eq!(grad_1.to_data(), Data::from([[7.0, 3.0], [4.0, 2.0]]));
         assert_eq!(grad_2.to_data(), Data::from([[2.0, 1.0], [3.0, 7.0]]));
     }
+
+    #[test]
+    fn should_diff_mask_scatter() {
+        let data_1 = Data::<f32, 2>::from([[1.0, 7.0], [2.0, 3.0]]);
+        let data_2 = Data::<f32, 2>::from([[4.0, 7.0], [2.0, 3.0]]);
+        let mask = Data::<bool, 2>::from([[true, false], [false, true]]);
+
+        let data_source = Data::<f32, 2>::from([[8.8, 8.8], [8.8, 8.8]]);
+
+        let tensor_1 = TestADTensor::from_data(data_1).require_grad();
+        let tensor_2 = TestADTensor::from_data(data_2).require_grad();
+        let tensor_source = TestADTensor::from_data(data_source);
```

Comment:
We should also validate the gradients for `tensor_source` with `require_grad()`. Also, did you validate the output values with a similar PyTorch program (just to be sure)?. I'm not sure of the backward pass and maybe your first implementation was right!

Reply:
Just let me have a try with PyTorch:blush:

Reply:
I've checked the gradients in PyTorch for the same scenario:

![image](https://user-images.githubusercontent.com/17194719/231419999-20a63b30-57d4-48e4-a175-8ed474afa7e6.png)

It seems the previous version for backward pass was right, so I revert it back.

Also the `tensor_source` has been labelled as `require_grad()` in its related test now.